jvb/src/main/java/org/jitsi/videobridge/Conference.java

```diff
@@ -413,42 +414,36 @@ private void lastNEndpointsChanged()
     /**
      * Notifies this instance that {@link #speechActivity} has identified a speaker switch event and there is now a new
      * dominant speaker.
+     * @param recentSpeakers the list of recent speakers (including the dominant speaker at index 0).
      */
-    private void dominantSpeakerChanged()
+    private void dominantSpeakerChanged(List<AbstractEndpoint> recentSpeakers)
     {
-        AbstractEndpoint dominantSpeaker = speechActivity.getDominantEndpoint();
-        String dominantSpeakerId = dominantSpeaker == null ? null : dominantSpeaker.getId();
-
+        List<String> recentSpeakersIds
+                = recentSpeakers.stream().map(AbstractEndpoint::getId).collect(Collectors.toList());
         if (logger.isInfoEnabled())
         {
-            logger.info("ds_change ds_id=" + dominantSpeakerId);
+            logger.info("Recent speakers changed: " + recentSpeakersIds);
             getVideobridge().getStatistics().totalDominantSpeakerChanges.increment();
         }
+        broadcastMessage(new DominantSpeakerMessage(recentSpeakersIds));
 
-        if (dominantSpeaker != null)
+        AbstractEndpoint dominantSpeaker = recentSpeakers.get(0);
+        if (getEndpointCount() > 2 && dominantSpeaker != null)
```

Comment:
Is it possible for the recentSpeakers list to change without the dominant speaker changing?  (E.g. if endpoints further down the list left?) If so, won't this send unnecessary keyframe requests?

Reply:
Yes, and yes. What would you think about removing the preemptive keyframe requests altogether?

Reply:
Ideally we'd only send preemptive keyframes when someone's actually likely to be moved into on-stage...with tile view being the default now, that's a lot less likely than it used to be.  But I'd be a bit reluctant to degrade the experience for that case.  I wonder how hard it is to distinguish...

Reply:
I pushed a fix to the regression you found.

 I'll follow up in another PR with sending this only when some local endpoint has someone "on-stage".